### ISSUE TITLE
Allow runtime changing of diff-so-fancy options by reading from ~/.gitconfig

### ISSUE THREAD

**Issue by scottchiefbaker** [so-fancy/diff-so-fancy#129]
Per @paulirish I'm creating a new issue for gitconfig stuff...

-----

I just posted a starter branch on my repo. Specifically the code would look something like:

https://github.com/scottchiefbaker/diff-so-fancy/blob/gitconfig/libs/header_clean/header_clean.pl#L10

and

https://github.com/scottchiefbaker/diff-so-fancy/blob/gitconfig/libs/header_clean/header_clean.pl#L171

It allows us to set run-time options to enable/disable certain behaviors. Still preliminary, but it's already reading changeHunkIndicators, stripLeadingSymbols, and markEmptyLines from your ~/.gitconfig. All these names are up for debate by the way, I just picked something simple.

**Comment by scottchiefbaker** [comment]
This would allow us to solve #69

**Comment by paulirish** [comment]
Nice. 

Is there convention for making sure all options that override default behavior must be set to "true"?  Wording is hard, but we may want to sort that out first.

**Comment by scottchiefbaker** [comment]
I was thinking the opposite. The current toggleable (is that a word) options are:

```
stripLeadingSymbols
markEmptyLines
changeHunkIndicators
```

All default to true, and you explicitly turn OFF the behavior? I can see it either way.

**Comment by paulirish** [comment]
Perhaps they might look something like...

```yaml
[diff-so-fancy]
   leadingSymbols = always
   emptyLineMarkers = never
   adjustHunkDisplay = never
```

And they default to 'auto'

**Comment by paulirish** [comment]
(revised the naming in my comment)

**Comment by scottchiefbaker** [comment]
My thinking was... without ANY config dsf performs as it does now (strips symbols, adjust hunks, etc). You "opted in" when you install dsf, in other words all the options default to "on" (true). If you want to disable one of the runtime options you add that to your config and set it to false.

The always/never syntax only makes sense to me if there is a third option (only in GUI, etc), otherwise we should just use true/false.

**Comment by paulirish** [comment]
k sounds good.

```yaml
[diff-so-fancy]
   stripLeadingSymbols = false
   markEmptyLines = false
   adjustHunkIndicators = false
```

**Comment by stevemao** [comment]
`markEmptyLines` shouldn't be an option I think... Whenever `stripLeadingSymbols` is `true`, `markEmptyLines` must be `true`. Vice versa.

**Comment by scottchiefbaker** [comment]
@stevemao `markEmptyLines` highlights in green or red any added/removed line that is just whitespace. Specifically an empty line, that's just a `\n`. That and `stripLeadingSymbols` aren't really connected.

Honestly, this whole thing started because I'm not a fan of `markEmptyLines` and I wanted a way to disable them :) But I **do** want to get rid of the leading `-` and `+`.

**Comment by stevemao** [comment]
> Honestly, this whole thing started because I'm not a fan of markEmptyLines and I wanted a way to disable them :) But I do want to get rid of the leading - and +.

You can't tell whether a line is added or removed then?

**Comment by scottchiefbaker** [comment]
In the example from our docs:

![Example](https://cloud.githubusercontent.com/assets/39191/13622719/7cc7c54c-e555-11e5-86c4-7045d91af041.png)

The forth line from the bottom has a red square indicating there was a removed (blank) line there. That's what I want to allow the usable to disable. That blank line indicator is not explicitly tied to removing the `+` and `-`. It's already working on my `gitconfig` repo if you want to check it out.

**Comment by scottchiefbaker** [comment]
Updated my `gitconfig` repo with all the latest bits merged, and everything else updated.

**Comment by stevemao** [comment]
> The forth line from the bottom has a red square indicating there was a removed (blank) line there. That's what I want to allow the user to disable. That blank line indicator is not explicitly tied to removing the + and -

So the user doesn't know if this empty is a removal or addition if there is no `+/-` nor colour.

**Comment by paulirish** [comment]
mostly closed by #141. followups filed elsewhere
